### PR TITLE
Feature/saga intl provider

### PR DESCRIPTION
**Pull request by cermakjiri**
__Changes__
- add MIT license
- update packages
- fix repo url in the package.json
- __Add getIntl saga for accessing the intl provider outside React context__

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,39 +7,44 @@
     "scripts": {
         "build:js": "babel src --out-dir lib --extensions \".ts,.tsx\" --source-maps inline",
         "build:types": "tsc --project tsconfig.types.json --emitDeclarationOnly",
-        "build": "npm run type-check && npm run build:js && npm run build:types",
-        "storybook": "start-storybook -p 6006",
+        "build": "npm run clean && npm run type-check && npm run build:js && npm run build:types",
+        "storybook": "rm -rf .storybook build && npm run build-storybook && start-storybook -p 6006",
```
Comment (horaklukas) [inline package.json]: proč maže konfiguraci storybooku `.storybook` ?

Comment (cermakjiri) [reply package.json]: to byla chybička, opraveno

File: src/HOC/translatableFactory.tsx [hunk 2]
```diff
@@ -1,43 +1,72 @@
...
 import { translateSelector } from '../services/selectors';
+import { createIntlProvider, destroyIntlProvider } from '../services/actions';
 
 interface WrappedProps extends LocaleState {
     [extraProps: string]: any;
 }
 
 const translatableFactory = (
-    intlLocaleData: LocaleData,
+    intlLocaleData: LocaleData<IntlLocaleData>,
```
Comment (horaklukas) [inline src/HOC/translatableFactory.tsx]: Why did you change this type? The previous one was correct..

File: package.json (hunk 1 above)
Comment (baierjak) [reply package.json]: i dont understand

File: src/HOC/translatableFactory.tsx (hunk 2 above)
Comment (horaklukas) [reply src/HOC/translatableFactory.tsx]: It wasn't wrong, but more general because the value of eg. `en` key has to be `object`, but according to your definition it could be `any`thing, even string, number, etc.

Comment (cermakjiri) [reply src/HOC/translatableFactory.tsx]: Ok, I see. I am still in a process of learning typescript, so thanks for the explanation.

File: src/HOC/translatableFactory.tsx [hunk 3]
```diff
@@ -1,43 +1,72 @@
...
+            }
+
+            render() {
+                const { locale } = this.props;
+                const intlProviderProps = {
+                    locale,
+                    key: locale,
+                    messages: intlLocaleData[locale],
+                    textComponent: Fragment,
+                };
```
Comment (horaklukas) [inline src/HOC/translatableFactory.tsx]: nice refactoring of the component 👏
